pre_commit/clientlib.py

```diff
@@ -222,6 +235,10 @@ def _entry(modname):
     ),
 
     MigrateShaToRev(),
+    cfgv.WarnAdditionalKeys(
```

Comment:
can you add a quick test to make sure this doesn't drift from the `CONFIG_REPO_DICT` definition?

Basically will need to iterate through the things in that `Map`'s `.items` and check the keys against the keys passed in here

Reply:
I'm not following. can you elaborate?

From what I understand you wish to add a test to validate that the new attribute does not change the definition of CONFIG_REPO_DICT and just added a kind of decorator. but I'm not sure how to do it.

Reply:
~something like this

```diff
--- a/pre_commit/clientlib.py
+++ b/pre_commit/clientlib.py
@@ -109,6 +109,8 @@ META = 'meta'
 
 
 class MigrateShaToRev(object):
+    key = 'rev'
+
     @staticmethod
     def _cond(key):
         return cfgv.Conditional(
diff --git a/tests/clientlib_test.py b/tests/clientlib_test.py
index cace0f3..13b42a5 100644
--- a/tests/clientlib_test.py
+++ b/tests/clientlib_test.py
@@ -305,3 +305,12 @@ def test_minimum_pre_commit_version_failing():
 def test_minimum_pre_commit_version_passing():
     cfg = {'repos': [], 'minimum_pre_commit_version': '0'}
     cfgv.validate(cfg, CONFIG_SCHEMA)
+
+
+@pytest.mark.parametrize('schema', (CONFIG_SCHEMA, CONFIG_REPO_DICT))
+def test_warn_additional(schema):
+    allowed_keys = {item.key for item in schema.items if hasattr(item, 'key')}
+    warn_additional, = [
+        x for x in schema.items if isinstance(x, cfgv.WarnAdditionalKeys)
+    ]
+    assert allowed_keys == set(warn_additional.keys)
```

Reply:
Done.
This test helped me find some more issues with the code. Thanks :-)